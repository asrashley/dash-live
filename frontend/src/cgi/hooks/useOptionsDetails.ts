import { type ReadonlySignal, useSignal } from "@preact/signals";

import { fieldGroups } from "@dashlive/options";

import { InputOptionName } from "../types/InputOptionName";

function createOptionNames(): InputOptionName[] {
    const names: InputOptionName[] = [];
    fieldGroups.forEach(grp => {
        grp.fields.forEach(({ name, fullName, shortName }) => {
            names.push({
                cgiName: name,
                fullName,
                shortName,
            });
        });
    });
    return names;
}

export interface UseOptionsDetailsHook {
    allOptions: ReadonlySignal<InputOptionName[]>;
}

export function useOptionsDetails(): UseOptionsDetailsHook {
    const allOptions = useSignal<InputOptionName[]>(createOptionNames());
    return { allOptions };
}