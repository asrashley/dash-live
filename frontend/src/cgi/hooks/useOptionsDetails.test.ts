import { renderHook } from "@testing-library/preact";
import { describe, expect, test } from "vitest";
import { useOptionsDetails } from "./useOptionsDetails";

describe('useOptionsDetails hook', () => {
    test('get all options', () => {
        const { result } = renderHook(() => useOptionsDetails());
        expect(result.current.allOptions.value).toMatchSnapshot();
    });
});