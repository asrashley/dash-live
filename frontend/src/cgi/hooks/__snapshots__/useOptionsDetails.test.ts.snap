// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useOptionsDetails hook > get all options 1`] = `
[
  {
    "cgiName": "abr",
    "fullName": "abr",
    "shortName": "ab",
  },
  {
    "cgiName": "acodec",
    "fullName": "audioCodec",
    "shortName": "ac",
  },
  {
    "cgiName": "events",
    "fullName": "eventTypes",
    "shortName": "evs",
  },
  {
    "cgiName": "timeline",
    "fullName": "segmentTimeline",
    "shortName": "st",
  },
  {
    "cgiName": "time",
    "fullName": "utcMethod",
    "shortName": "utc",
  },
  {
    "cgiName": "player",
    "fullName": "videoPlayer",
    "shortName": "vp",
  },
  {
    "cgiName": "ad_audio",
    "fullName": "audioDescription",
    "shortName": "ad",
  },
  {
    "cgiName": "start",
    "fullName": "availabilityStartTime",
    "shortName": "ast",
  },
  {
    "cgiName": "bugs",
    "fullName": "bugCompatibility",
    "shortName": "bug",
  },
  {
    "cgiName": "drift",
    "fullName": "clockDrift",
    "shortName": "dft",
  },
  {
    "cgiName": "leeway",
    "fullName": "leeway",
    "shortName": "lee",
  },
  {
    "cgiName": "main_audio",
    "fullName": "mainAudio",
    "shortName": "ma",
  },
  {
    "cgiName": "main_text",
    "fullName": "mainText",
    "shortName": "mt",
  },
  {
    "cgiName": "mup",
    "fullName": "minimumUpdatePeriod",
    "shortName": "mup",
  },
  {
    "cgiName": "ntp_servers",
    "fullName": "ntpSources",
    "shortName": "ntps",
  },
  {
    "cgiName": "tcodec",
    "fullName": "textCodec",
    "shortName": "tc",
  },
  {
    "cgiName": "tlang",
    "fullName": "textLanguage",
    "shortName": "tl",
  },
  {
    "cgiName": "base",
    "fullName": "useBaseUrls",
    "shortName": "base",
  },
  {
    "cgiName": "patch",
    "fullName": "patch",
    "shortName": "patch",
  },
  {
    "cgiName": "depth",
    "fullName": "timeShiftBufferDepth",
    "shortName": "tbd",
  },
  {
    "cgiName": "clearkey__enabled",
    "fullName": "enabled",
    "shortName": "clearkey__enabled",
  },
  {
    "cgiName": "clearkey__la_url",
    "fullName": "licenseUrl",
    "shortName": "clu",
  },
  {
    "cgiName": "clearkey__drmloc",
    "fullName": "drmloc",
    "shortName": "clearkey__drmloc",
  },
  {
    "cgiName": "marlin__enabled",
    "fullName": "enabled",
    "shortName": "marlin__enabled",
  },
  {
    "cgiName": "marlin__la_url",
    "fullName": "licenseUrl",
    "shortName": "mlu",
  },
  {
    "cgiName": "marlin__drmloc",
    "fullName": "drmloc",
    "shortName": "marlin__drmloc",
  },
  {
    "cgiName": "ping__count",
    "fullName": "count",
    "shortName": "pinCoun",
  },
  {
    "cgiName": "ping__duration",
    "fullName": "duration",
    "shortName": "pinDura",
  },
  {
    "cgiName": "ping__inband",
    "fullName": "inband",
    "shortName": "pinInba",
  },
  {
    "cgiName": "ping__interval",
    "fullName": "interval",
    "shortName": "pinInte",
  },
  {
    "cgiName": "ping__start",
    "fullName": "start",
    "shortName": "pinStar",
  },
  {
    "cgiName": "ping__timescale",
    "fullName": "timescale",
    "shortName": "pinTime",
  },
  {
    "cgiName": "ping__value",
    "fullName": "value",
    "shortName": "pinValu",
  },
  {
    "cgiName": "ping__version",
    "fullName": "version",
    "shortName": "pinVers",
  },
  {
    "cgiName": "playready__enabled",
    "fullName": "enabled",
    "shortName": "playready__enabled",
  },
  {
    "cgiName": "playready__la_url",
    "fullName": "licenseUrl",
    "shortName": "plu",
  },
  {
    "cgiName": "playready__piff",
    "fullName": "piff",
    "shortName": "pff",
  },
  {
    "cgiName": "playready__version",
    "fullName": "version",
    "shortName": "pvn",
  },
  {
    "cgiName": "playready__drmloc",
    "fullName": "drmloc",
    "shortName": "playready__drmloc",
  },
  {
    "cgiName": "scte35__count",
    "fullName": "count",
    "shortName": "sctCoun",
  },
  {
    "cgiName": "scte35__duration",
    "fullName": "duration",
    "shortName": "sctDura",
  },
  {
    "cgiName": "scte35__inband",
    "fullName": "inband",
    "shortName": "sctInba",
  },
  {
    "cgiName": "scte35__interval",
    "fullName": "interval",
    "shortName": "sctInte",
  },
  {
    "cgiName": "scte35__program_id",
    "fullName": "program_id",
    "shortName": "sctProg",
  },
  {
    "cgiName": "scte35__start",
    "fullName": "start",
    "shortName": "sctStar",
  },
  {
    "cgiName": "scte35__timescale",
    "fullName": "timescale",
    "shortName": "sctTime",
  },
  {
    "cgiName": "scte35__value",
    "fullName": "value",
    "shortName": "sctValu",
  },
  {
    "cgiName": "scte35__version",
    "fullName": "version",
    "shortName": "sctVers",
  },
]
`;
